File: Módulo-C/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    //solicitar texto ao usuário
    string texto = get_string("Texto: ");

    //contar o número de letras que há no texto
    int letras = 0;
    for (int i = 0, n = strlen(texto); i < n; i ++) // strlen = calcula o comprimento de uma string
    {
        if (isalpha(texto[i])) // verificar se um caractere é alfabético
        {
            letras ++;
        }
    }

    //contar o número de palavras
    int palavras = 1;
    for (int i = 0, n = strlen(texto); i < n; i ++)
    {
        if (isspace(texto[i])) //verifique se um caractere é um espaço em branco
        {
            palavras ++;  
        }
        
    }

    //contar o número de sentenças
    int sentences = 0;
    for (int i = 0, n = strlen(texto); i < n; i ++)
    {
        if (texto[i] == '.' || texto[i] == '!' || texto[i] == '?')
        {
            sentences ++;
        }
    }
    
    //Usando índice Coleman-Liau para determinar o nível de leitura
    float calculation = (0.0588 * letras / palavras * 100) - 
                        (0.296 * sentences / palavras * 100) -
                        15.8;
    int index = round(calculation);
    
    //Verificando o nível de leitura
    if (index < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: Linguagem-Python/sentimental-readability/readability.py
# TODO

from cs50 import get_string

# solicitar texto ao usuário
text = get_string("Text: ")

letters = 0
words = 1
sentences = 0

# Contar número de letras, palavras e sentenças
for i in text:
    if i.isalpha():
        letters += 1
    elif i == " ":
        words += 1
    elif i == "." or i == "!" or i == "?":
        sentences += 1

# índice coleman-liau
index = 0.0588 * (letters/words*100) - 0.296 * (sentences /
                                                words*100) - 15.8

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print("Grade ", round(index))